lib/tesla/adapter/gun.ex

```diff
@@ -118,7 +134,12 @@ if Code.ensure_loaded?(:gun) do
     defp open_conn(url, opts) do
       uri = URI.parse(url)
       opts = if uri.scheme == "https", do: Map.put(opts, :transport, :tls), else: opts
-      {:ok, pid} = :gun.open(to_charlist(uri.host), uri.port, Map.take(opts, @gun_keys))
+
+      {:ok, pid} =
+        if opts[:conn],
+          do: {:ok, opts[:conn]},
+          else: :gun.open(to_charlist(uri.host), uri.port, Map.take(opts, @gun_keys))
+
```

Comment:
I'd go with standard multiline `if` ;)

Reply:
In new version it's made with [clause](https://github.com/teamon/tesla/pull/310/files#diff-f93babd4700a230f6ced434c9dd3a68eR148)